### PR TITLE
Cluster controller unit test cleanup, part 2 [run-systemtest] 

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java [hunk 1]
```diff
@@ -1193,17 +1197,17 @@ public void waitForNodesHavingSystemStateVersionEqualToOrAbove(int version, int
                     context.log(logger, Level.INFO, ackedNodes + " nodes now have acked system state " + version + " or higher.");
                     return;
                 }
-                long remainingTime = maxTime - System.currentTimeMillis();
-                if (remainingTime <= 0) {
-                    throw new IllegalStateException("Did not get " + nodeCount + " nodes to system state " + version + " within timeout of " + timeout + " milliseconds.");
+                Duration remainingTime = Duration.between(Instant.now(), endTime);
+                if (remainingTime.isNegative()) {
```
Comment (baldersheim) [inline clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java:1201]: Prefer Instant.now().isAfter(endTime) as above for consistency and readability.

File: clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java [hunk 2]
```diff
@@ -1215,10 +1219,10 @@ public void waitForNodesInSlobrok(int distNodeCount, int storNodeCount, int time
                 }
                 if (distCount == distNodeCount && storCount == storNodeCount) return;
 
-                long remainingTime = maxTime - System.currentTimeMillis();
-                if (remainingTime <= 0) {
+                Duration remainingTime = Duration.between(Instant.now(), endTime);
+                if (remainingTime.isNegative()) {
```
Comment (baldersheim) [inline clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java:1223]: Prefer Instant.now().isAfter(endTime) as above for consistency and readability.

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/DatabaseTest.java [hunk 3]
```diff
@@ -154,7 +154,7 @@ private void setWantedState(Node n, NodeState ns, Map<Node, NodeState> wantedSta
         Request req = new Request("setNodeState");
         req.parameters().add(new StringValue("storage/cluster.mycluster/" + n.getType().toString() + "/" + n.getIndex()));
         req.parameters().add(new StringValue(ns.serialize(true)));
-        connection.invokeSync(req, timeoutS);
+        connection.invokeSync(req, timeout.getSeconds());
```
Comment (baldersheim) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/DatabaseTest.java:157]: getSeconds will round down. 3.5 => 3, 0.9 => 0.
That might be fine here in the test, but generally it is not. Can Duration be converted to a double in seconds ?

**Review by baldersheim — CHANGES_REQUESTED** [review]
Consider the comments. Perhaps the invoke methods should take a a Duration instead, or in addition.

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/DatabaseTest.java (hunk 3 above)
Comment (hmusum) [reply clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/DatabaseTest.java]: Good point, will fix

File: clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java (hunk 2 above)
Comment (hmusum) [reply clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java]: 👍

File: clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java (hunk 1 above)
Comment (hmusum) [reply clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/FleetController.java]: 👍

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/RpcServerTest.java [hunk 4]
```diff
@@ -120,15 +119,15 @@ public String isConditionMet() {
...
+        }, null, timeout());
 
         int rpcPort = fleetController.getRpcPort();
         supervisor = new Supervisor(new Transport());
         Target connection = supervisor.connect(new Spec("localhost", rpcPort));
         assertTrue(connection.isValid());
 
         Request req = new Request("getSystemState");
-        connection.invokeSync(req, timeoutS);
+        connection.invokeSync(req, timeout().getSeconds());
```
Comment (baldersheim) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/RpcServerTest.java:130]: double timeoutInSeconds() ?

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/FleetControllerTest.java [hunk 5]
```diff
@@ -479,7 +460,7 @@ void setWantedState(DummyVdsNode node, State state, String reason) {
         Request req = new Request("setNodeState");
         req.parameters().add(new StringValue(node.getSlobrokName()));
         req.parameters().add(new StringValue(ns.serialize()));
-        connection.invokeSync(req, timeoutS);
+        connection.invokeSync(req, timeout.getSeconds());
```
Comment (baldersheim) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/FleetControllerTest.java:463]: double timeoutInSeconds() ?

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java [hunk 6]
```diff
@@ -312,13 +307,13 @@ void testMasterZooKeeperCooldown() throws Exception {
...
-        long endTime = System.currentTimeMillis() + timeoutMS;
-        while (System.currentTimeMillis() < endTime) {
+    private void waitForMasterReason(String reason, Integer master, List<Target> connections, int[] nodes) {
+        Instant endTime = Instant.now().plus(timeout());
+        while (Instant.now().isBefore(endTime)) {
             boolean allOk = true;
             for (int node : nodes) {
                 Request req = new Request("getMaster");
-                connections.get(node).invokeSync(req, FleetControllerTest.timeoutS);
+                connections.get(node).invokeSync(req, timeout().getSeconds());
```
Comment (baldersheim) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java:316]: double timeoutInSeconds() ?

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java [hunk 7]
```diff
@@ -362,9 +356,9 @@ void testGetMaster() throws Exception {
         Request req = new Request("getMaster");
 
         for (int nodeIndex = 0; nodeIndex < 3; ++nodeIndex) {
-            for (int retry = 0; retry < FleetControllerTest.timeoutS * 10; ++retry) {
+            for (int retry = 0; retry < timeout().getSeconds() * 10; ++retry) {
```
Comment (baldersheim) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java:359]: double timeoutInSeconds() ?

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java [hunk 8]
```diff
@@ -395,23 +389,23 @@ void testGetMaster() throws Exception {
         waitForMaster(1);
 
         req = new Request("getMaster");
-        connections.get(0).invokeSync(req, FleetControllerTest.timeoutS);
+        connections.get(0).invokeSync(req, timeoutInSeconds());
         assertEquals(104, req.errorCode(), req.toString());
         assertEquals("Connection error", req.errorMessage(), req.toString());
 
-        for (int i = 0; i < FleetControllerTest.timeoutS * 10; ++i) {
+        for (int i = 0; i < timeout().getSeconds() * 10; ++i) {
```
Comment (baldersheim) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java:396]: double timeoutInSeconds() ?

File: clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java [hunk 9]
```diff
@@ -395,23 +389,23 @@ void testGetMaster() throws Exception {
...
+            connections.get(1).invokeSync(req, timeoutInSeconds());
             assertFalse(req.isError(), req.errorMessage());
             if (req.returnValues().get(0).asInt32() != -1) break;
             // We may have bad timing causing node not to have realized it is master yet
         }
         assertEquals(1, req.returnValues().get(0).asInt32(), req.toString());
         assertEquals("2 of 3 nodes agree 1 is master.", req.returnValues().get(1).asString(), req.toString());
 
-        for (int i = 0; i < FleetControllerTest.timeoutS * 10; ++i) {
+        for (int i = 0; i < timeout().getSeconds() * 10; ++i) {
```
Comment (baldersheim) [inline clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/MasterElectionTest.java:406]: double timeoutInSeconds() ?

**Review by baldersheim — COMMENTED** [review]
Still 6 more getSeconds()

**Comment by hmusum** [comment]
All should be fixed now

**Review by baldersheim — APPROVED** [review]

**Pull request merged**